addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCDigitalBlockHandler.java

```diff
@@ -0,0 +1,262 @@
+ */
+package org.openhab.binding.plclogo.handler;
+
+import static org.openhab.binding.plclogo.PLCLogoBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.openhab.binding.plclogo.config.PLCLogoDigitalConfiguration;
+import org.openhab.binding.plclogo.internal.PLCLogoDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import Moka7.S7;
+
+/**
+ * The {@link PLCDigitalBlockHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Falkenstern - Initial contribution
+ */
+public class PLCDigitalBlockHandler extends PLCBlockHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(PLCDigitalBlockHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_DIGITAL);
+
+    private PLCLogoDigitalConfiguration config = getConfigAs(PLCLogoDigitalConfiguration.class);
+    private int oldValue = Integer.MAX_VALUE;
+
+    public PLCDigitalBlockHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        final Thing thing = getThing();
+        Objects.requireNonNull(thing, "PLCDigitalBlockHandler: Thing may not be null.");
+
+        final Bridge bridge = getBridge();
+        Objects.requireNonNull(bridge, "PLCDigitalBlockHandler: Bridge may not be null.");
+
+        synchronized (config) {
+            config = getConfigAs(PLCLogoDigitalConfiguration.class);
+        }
+
+        final String name = config.getBlockName();
+        logger.debug("Initialize LOGO! {} digital handler.", name);
+
+        scheduler.execute(new Runnable() {
+            @Override
+            public void run() {
+                if (config.isBlockValid()) {
+                    ThingBuilder tBuilder = editThing();
+
+                    String text = config.isInputBlock() ? INPUT_CHANNEL : OUTPUT_CHANNEL;
+                    text = text.substring(0, 1).toUpperCase() + text.substring(1);
+                    tBuilder = tBuilder.withLabel(bridge.getLabel() + ": " + text + " " + name);
```

Comment:
This looks familiar, same comments apply (the builders and maybe more :-) ), please check whether there is possibility to share some code in a smart way.